internal/types/compare_order.go

```diff
@@ -253,6 +292,31 @@ func compareTypeOrder(a, b any) CompareResult {
 	}
 }
 
+// compareType compares compareTypeOrder then compares
+// number types. The number types int32, int64 and float64
+// are considered different.
+func compareType(a, b any) CompareResult {
+	if res := compareTypeOrder(a, b); res != Equal {
+		return res
+	}
+
+	if detectDataType(a) == numbersDataType {
+		aType := detectNumberType(a)
+		bType := detectNumberType(b)
+
+		switch {
+		case aType < bType:
+			return Less
+		case aType > bType:
+			return Greater
+		default:
+			return Equal
```

Comment:
Are there any use cases for it? If I understand the task correctly, it's enough to know whether the types are equal or not.

Reply:
No longer applicable now. We use separate func Identical for this work